core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -642,16 +667,21 @@ class Partition(val topicPartition: TopicPartition,
    *                   topic partition
    * @param isr The set of broker ids that are known to be insync with the leader
    */
-  def updateAssignmentAndIsr(assignment: Seq[Int], isr: Set[Int]): Unit = {
+  def updateAssignmentAndIsr(assignment: Seq[Int],
+                             isr: Set[Int],
+                             addingReplicas: Seq[Int] = Seq.empty,
```

Comment:
nit: do these need to be optional? Usually we favor explicit arguments outside of test code.

Reply:
Not necessarily, I can refactor it.